app/routes.py

```diff
@@ -1,2 +1,323 @@
 from flask import Blueprint
+from app.models.Videos import Video 
+from app.models.Customers import Customer
+from app.models.Rentals import Rental
+from flask import Blueprint, make_response, jsonify, request
+from app import db 
+from sqlalchemy import asc, desc
+from datetime import datetime, date, timedelta
+import requests 
+import os 
+
+#blueprints 
+customer_bp = Blueprint("customers", __name__, url_prefix="/customers")
+
+video_bp = Blueprint("videos", __name__, url_prefix="/videos")
+
+rental_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+
+#helper functions 
+def make_400():
+    return make_response({"Oops! Something went wrong. ": "Invalid data or format."}, 400)
+
+def make_404():
+    return make_response("Whatever you are looking for, we didn't find it.", 404)
+
+def update_customer_from_json(body):
+    pass
+
+def get_due_date():
+    due_date = date.today() + timedelta(days=7)
+    return due_date
+
+def get_rental_by_customer(customer_id):
+    customer_info = db.session.query(Customer, Video, Rental).join(Customer, Customer.customer_id==Rental.customer_id)\
+            .join(Video, Video.video_id==Rental.video_id).filter(Customer.customer_id == customer_id).all()
+    return customer_info
+
+def get_rental_by_video(video_id):
+    video_info = db.session.query(Customer, Video, Rental).join(Customer, Customer.customer_id==Rental.customer_id)\
+            .join(Video, Video.video_id==Rental.video_id).filter(Video.video_id==video_id).all()
+    return video_info
+
+
+#/customer endpoints 
+@customer_bp.route("", methods=["GET"])
+def get_all_customers():
+    sort_query = request.args.get("sort")
+
+    if sort_query:
+        if sort_query == "name":
+            customers = Customer.query.order_by(asc(Customer.customer_name))
+        elif sort_query == "postal_code":
+            customers = Customer.query.order_by(asc(Customer.customer_zip))
+        elif sort_query == "registered_at":
+            customers = Customer.query.order_by(asc(Customer.register_at))
+        elif sort_query == "id":
+            customers = Customer.query.order_by(asc(Customer.customer_id))
+        else:
+            return make_400()
+    else:
+        customers = Customer.query.all()
+    customer_response = []
+    for customer in customers:
+        customer_response.append(customer.to_json())    
+    return jsonify(customer_response), 200
+
+@customer_bp.route("", methods=["DELETE"])
+def delete_all_customers():
+    customers = Customer.query.all()
+    for customer in customers:
+        db.session.delete(customer)
+    db.session.commit()
+    return make_response("All of the customers in the database have been deleted.", 200)
+
+@customer_bp.route("/<customer_id>", methods=["GET"])
+def get_single_customer(customer_id):
+    customer = Customer.query.get(customer_id)
```

Comment:
You can use `get_or_404` here:

```suggestion
    customer = Customer.query.get_or_404(customer_id)
```